### PR TITLE
Fix wrong recognition of Currency pieces as DateTime

### PR DESCRIPTION
**Pull request by guom08**
There is another known issue related to recognition "." splitter expression as DateTime.
"5.1938" is currently recognized as "year 1938" in all languages
"5.11" is ambiguous and needs some prepositions to disambiguate.
The fix will come in another PR related to issue # 908

The fix in this PR also contains some Regex refactor to English DateTime Regex.
This refactor also applied to Spanish, Portuguese, French and German.
For Spanish and Portuguese, this refactor fixed the wrong recognition for the currency as DateTime entity.
For French and German, this refactor doesn't fix all the cases as in French and German. This is because there is an existing issue to recognize only "5" and "00" as Time entity. Fix for this issue will be in an upcoming PR.

Also this PR include two small fixes.
1. For cases like "at 6.45 percent", "at 6.45" should not be recognized as a Time entity
2. For cultureCode like "es-mx", map to the nearest language "es-es"
3. Fix cases like "two month ago and one month ago", to return two DateRange entities separately.

### CODE REVIEW

File: .NET/Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishDateExtractorConfiguration.cs [hunk 1]
```diff
@@ -62,15 +62,15 @@ public class SpanishDateExtractorConfiguration : BaseOptionsConfiguration, IDate
             new Regex(DateTimeDefinitions.RelativeWeekDayRegex, RegexOptions.IgnoreCase | RegexOptions.Singleline);
 
         //TODO: modify below regex according to the counterpart in English
-        public static readonly Regex ForTheRegex = new Regex($@"^[.]",
+        public static readonly Regex ForTheRegex = new Regex($@".^",
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishDateExtractorConfiguration.cs]: Shouldn't these point to the YAML file?

File: .NET/Microsoft.Recognizers.Definitions/Spanish/DateTimeDefinitions.cs [hunk 2]
```diff
@@ -73,14 +74,14 @@ public static class DateTimeDefinitions
...
-		public static readonly string DateExtractor2 = $@"\b({WeekDayRegex}(\s+|\s*,\s*))?{DayRegex}\s*([\.\-]|de)?\s*{MonthRegex}?(\s*,\s*|\s*(del?)\s*){YearRegex}\b";
-		public static readonly string DateExtractor3 = $@"\b({WeekDayRegex}(\s+|\s*,\s*))?{DayRegex}(\s+|\s*,\s*|\s+de\s+|\s*-\s*){MonthRegex}((\s+|\s*,\s*){YearRegex})?\b";
-		public static readonly string DateExtractor4 = $@"\b{MonthNumRegex}\s*[/\\\-]\s*{DayRegex}\s*[/\\\-]\s*{YearRegex}";
-		public static readonly string DateExtractor5 = $@"\b{DayRegex}\s*[/\\\-\.]\s*{MonthNumRegex}\s*[/\\\-\.]\s*{YearRegex}";
+		public static readonly string DateExtractor2 = $@"\b({WeekDayRegex}(\s+|\s*,\s*))?{DayRegex}\s*([\.\-]|de)\s*{MonthRegex}(\s*,\s*|\s*(del?)\s*)({YearRegex}|{TwoDigitYearRegex})\b";
+		public static readonly string DateExtractor3 = $@"\b({WeekDayRegex}(\s+|\s*,\s*))?{DayRegex}(\s+|\s*,\s*|\s+de\s+|\s*-\s*){MonthRegex}((\s+|\s*,\s*)({YearRegex}|{TwoDigitYearRegex}))?\b";
+		public static readonly string DateExtractor4 = $@"\b{MonthNumRegex}\s*[/\\\-]\s*{DayRegex}\s*[/\\\-]\s*({YearRegex}|{TwoDigitYearRegex})";
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Definitions/Spanish/DateTimeDefinitions.cs]: Is this similar to how the issues is handled in English? Ideally it should be solved in the same way (if possible).
Also, this fix needs to propagate across languages.

File: Patterns/Spanish/Spanish-DateTime.yaml [hunk 3]
```diff
@@ -8,7 +8,9 @@ DayRegex: !simpleRegex
 MonthNumRegex: !simpleRegex
   def: (?<month>01|02|03|04|05|06|07|08|09|10|11|12|1|2|3|4|5|6|7|8|9)\b
 YearRegex: !simpleRegex
-  def: (?<year>((1[5-9]|20)\d{2})|[0-27-9]\d)
+  def: \b(?<year>(1[5-9]|20)\d{2})(?!\.0\b)\b
```
Comment (tellarin) [inline Patterns/Spanish/Spanish-DateTime.yaml]: Can you add a comment in the YAML file here to explain (?!\.0\b)?
Also, this fix also should propagate across languages.

File: Specs/DateTime/English/DateTimeModel.json [hunk 4]
```diff
@@ -7364,5 +7364,44 @@
         }
       }
     ]
+  },
+  {
+    "Input": "Will he be back in two days? or in a week?",
+    "Context": {
+      "ReferenceDateTime": "2018-10-17T12:00:00"
+    },
+    "NotSupported": "java, javascript, python",
```
Comment (tellarin) [inline Specs/DateTime/English/DateTimeModel.json]: javascript?

File: Specs/DateTime/English/DateTimeModelCalendarMode.json [hunk 5]
```diff
@@ -898,42 +898,5 @@
         }
       }
     ]
-  },
-  {
-    "Input": "Mortgages were at 6.45 percent",
```
Comment (tellarin) [inline Specs/DateTime/English/DateTimeModelCalendarMode.json]: Why delete these specs?

File: Specs/DateTime/English/DateTimeModelComplexCalendar.json [hunk 6]
```diff
@@ -6272,5 +6309,44 @@
         }
       }
     ]
+  },
+  {
+    "Input": "Will he be back in two days? or in a week?",
+    "Context": {
+      "ReferenceDateTime": "2018-10-17T12:00:00"
+    },
+    "NotSupported": "java, javascript, python",
```
Comment (tellarin) [inline Specs/DateTime/English/DateTimeModelComplexCalendar.json]: javascript?

File: Specs/DateTime/English/DateTimeModelCalendarMode.json (hunk 5 above)
Comment (guom08) [reply Specs/DateTime/English/DateTimeModelCalendarMode.json]: These should be in the ComplexCalendar spec.

**Pull request merged**